ambry-store/src/main/java/com/github/ambry/store/BlobStoreStats.java

```diff
@@ -160,6 +164,12 @@ static StatsSnapshot convertStoreUsageToContainerStatsSnapshot(Map<Short, Map<Sh
       queueProcessor = new QueueProcessor();
       shortLiveTaskScheduler.scheduleAtFixedRate(queueProcessor, 0, queueProcessingPeriodInMs, TimeUnit.MILLISECONDS);
     }
+    if (enableAsyncGetValidSize) {
+      ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
```

Comment:
Here we are using a single thread per each blob store. This could be a lot of extra threads that will be mostly idle. Could we reuse the shared `longLiveTaskScheduler` or `shortLiveTaskScheduler`? (not exactly sure when a task is considered to be short vs long, but we can look into that.

Reply:
I used the shortLiveTaskScheduler for my case. longLiveTaskScheduler only have 1 thread in thread pool it's running less frequently.